src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectSnapshotChangeTrigger.cs

```diff
@@ -23,51 +23,60 @@ private void InitializeSolution(Solution solution)
         {
             Debug.Assert(solution != null);
 
-            _projectManager.ProjectsCleared();
+            _projectManager.WorkspaceProjectsCleared();
 
             foreach (var project in solution.Projects)
             {
                 if (project.Language == LanguageNames.CSharp)
                 {
-                    _projectManager.ProjectAdded(project);
+                    _projectManager.WorkspaceProjectAdded(project);
                 }
             }
         }
 
         // Internal for testing
         internal void Workspace_WorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
         {
-            Project underlyingProject;
+            Project project;
             switch (e.Kind)
             {
                 case WorkspaceChangeKind.ProjectAdded:
                     {
-                        underlyingProject = e.NewSolution.GetProject(e.ProjectId);
-                        Debug.Assert(underlyingProject != null);
+                        project = e.NewSolution.GetProject(e.ProjectId);
+                        Debug.Assert(project != null);
 
-                        if (underlyingProject.Language == LanguageNames.CSharp)
+                        if (DefaultProjectSnapshotManager.IsSupportedWorkspaceProject(project))
```

Comment:
The snapshot manager checks this anyways. I'd un-static the method and leave that checking to the manager. Same elsewhere

Reply:
Sure